### PR TITLE
Upon Second Schema deployment, VersionHistoryService reports error and

### PR DESCRIPTION
**Pull request by prb112**
the update-schema is stopped. #1065

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java [hunk 1]
```diff
@@ -893,7 +893,7 @@ protected void checkIfTenantNameAndTenantKeyExists(Db2Adapter adapter, String te
                 final String sql =
                         "SELECT t.tenant_status FROM fhir_admin.tenants t WHERE t.tenant_name = ? "
                                 + "AND EXISTS (SELECT 1 FROM fhir_admin.tenant_keys tk WHERE tk.mt_id = t.mt_id "
-                                + "AND tk.tenant_hash = sysibm.hash(tk.tenant_salt || ?, 2));";
+                                + "AND tk.tenant_hash = sysibm.hash(tk.tenant_salt || ?, 2))";
```
Comment (prb112) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:896]: DB2 SQL Error: SQLCODE=-104, SQLSTATE=42601, SQLERRMC=;;enant_salt || ?, 2));END-OF-STATEMENT
Possibly double end of statement problem? maybe only occurs on Windows versus Mac?

Comment (lmsurpre) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:897]: wouldn't our windows e2e test have caught this?  if not, why not?

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/DerbyBootstrapper.java [hunk 2]
```diff
@@ -111,12 +111,12 @@ public static void bootstrap(Connection connection, String adminSchemaName, Stri
...
         // to determine which updates to apply and to record the new changes as they
         // are applied
-        VersionHistoryService vhs = new VersionHistoryService(adminSchemaName, dataSchemaName);
+        VersionHistoryService vhs = new VersionHistoryService(adminSchemaName, dataSchemaName, OAUTH_SCHEMANAME);
         vhs.setTarget(adapter);
         vhs.init();
 
         // Use the version history service to determine if this table existed before we run `applyWithHistory`
-        boolean newDb = vhs.getVersion(dataSchemaName, DatabaseObjectType.TABLE.name(), "PARAMETER_NAMES") == null;
+        boolean newDb = vhs.getVersion(dataSchemaName, DatabaseObjectType.TABLE.name(), "PARAMETER_NAMES") == 0;
```
Comment (lmsurpre) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/DerbyBootstrapper.java:119]: should it be `newDb = (x == null || x == 0)`?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java (hunk 1 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:897]: I thought so too. 

I did send a new cli jar to Frances and it worked.

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/DerbyBootstrapper.java (hunk 2 above)
Comment (prb112) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/DerbyBootstrapper.java]: fixed

**Pull request merged**